### DISCUSSION TITLE
Redirecting to an android app

### DISCUSSION THREAD

**Discussion by mackatap** [GodotNuts/GodotFirebase#287] [Q&A]
Hello godotnuts team!
What do I need to get the browser on android to redirect back to my app after creating a google auth token? I own a domain that I could use if needed. I would like to just use standard google auth tokens.  Could I get any pointers in the right direction? I assume I will use get_google_auth_redirect(redirect_uri : String, listen_to_port : int)
Thank you!

**Comment by luiscesjr** [comment]
Hi! I think this will be of help:

https://stackoverflow.com/questions/33870983/redirect-user-from-browser-to-my-app-after-open-a-specific-url

**Reply by BearDooks** [reply]
Dynamic Links is part of the plugin that we publish, but as of this moment I am not sure of to many people who are using them, so this might open another can of worms that we would need to work through. More than happy to help if that is the case

**Reply by mackatap** [reply]
I would still really like to do this to avoid the copy and paste. I noticed get_google_auth_redirect was removed. How can I use dynamic links? get_google_auth_localhost() does what I'd like, but how do i get it to go back to my app instead of a 404 page?
Thanks for any help!

**Reply by mackatap** [reply]
I have just learned that the OAuth OOB method (manual copy/paste) will be deprecated by google on October 3, 2022. This seems more important now. Could really use some help, thank you!

**Reply by fenix-hub** [reply]
You don't need manual method, I've deprecated that some versions ago :)
You can use [this one](https://github.com/GodotNuts/GodotFirebase/blob/4df1a40144e9c1ecba85f6abf3021057bd33ec5a/addons/godot-firebase/auth/auth.gd#L263) instead.

**Reply by mackatap** [reply]
Ok thank you! Once I use that function, how do I access the token so I can login with it? I assume I would use login_with_oauth(token, provider).

**Reply by fenix-hub** [reply]
@mackatap 
You don't need to call that function manually. It will be called automatically by our plugin and OAuth2 will consider you "already logged in" once users grant permissions to access their accounts.
I'll write more in the docs, but the flow is pretty much this:

```gdscript
func _ready() -> void:
    Firebase.Auth.connect("login_succeeded", self, "_on_login") 
    # Chose your preferred provider, make sure to set up .env file accordingly
    var provider: AuthProvider = Firebase.Auth.get_GoogleProvider()
    # Make an auth request.  You can make it on localhost if on desktop, 
    # but the preferred way on all platforms is with a redirect to your app/website
    # Firebase.Auth.get_auth_localhost(provider)
    Firebase.Auth.set_redirect_uri("< YOUR WEBSITE OR DYNAMIC LINK TO APP >") # this one should be configured on your oauth provider console as the `redirect_uri`
    Firebase.Auth.get_auth_with_redirect(provider)

func _on_login(login: Dictionary) -> void:
    print(login) # this will print the `auth` body. Inside this dictionary there will be an `idToken` or `accessToken`. It is a JWT
```
